### ISSUE TITLE
Issue with running Uchiwa out of the box

### ISSUE THREAD

**Issue by umair-hussain-hs** [sensu/uchiwa#518]
Currently, Uchiwa uses Jwt-Go from https://github.com/dgrijalva/jwt-go. 

Running the following setup command (listed in the docs):
`$ go get github.com/sensu/uchiwa`
Produces the following error:
```
# github.com/sensu/uchiwa/uchiwa/auth
go/src/github.com/sensu/uchiwa/uchiwa/auth/handlers.go:41: undefined: jwt.ParseFromRequest
go/src/github.com/sensu/uchiwa/uchiwa/auth/roles.go:22: invalid operation: t.Claims["Role"] (type jwt.Claims does not support indexing)
go/src/github.com/sensu/uchiwa/uchiwa/auth/token.go:26: invalid operation: t.Claims["Role"] (type jwt.Claims does not support indexing)
go/src/github.com/sensu/uchiwa/uchiwa/auth/token.go:27: invalid operation: t.Claims["Username"] (type jwt.Claims does not support indexing)
```

Upon closer inspection, we see that the issue lies with the repo mentioned above. About 14 days ago, version 3.0.0 was released and part of the changes pertained to the function mentioned in the error log above.

Looking at that repos version history (https://github.com/dgrijalva/jwt-go/blob/master/VERSION_HISTORY.md) we find the following relevant changes:
```
## `jwt-go` Version History

#### 3.0.0

* **Compatibility Breaking Changes**: See MIGRATION_GUIDE.md for tips on updating your code
	* Dropped support for `[]byte` keys when using RSA signing methods.  This convenience feature could contribute to security vulnerabilities involving mismatched key types with signing methods.
	* `ParseFromRequest` has been moved to `request` subpackage and usage has changed
	* The `Claims` property on `Token` is now type `Claims` instead of `map[string]interface{}`.  The default value is type `MapClaims`, which is an alias to `map[string]interface{}`.  This makes it possible to use a custom type when decoding claims.
* Other Additions and Changes
	* Added `Claims` interface type to allow users to decode the claims into a custom type
	* Added `ParseWithClaims`, which takes a third argument of type `Claims`.  Use this function instead of `Parse` if you have a custom type you'd like to decode into.
	* Dramatically improved the functionality and flexibility of `ParseFromRequest`, which is now in the `request` subpackage
	* Added `ParseFromRequestWithClaims` which is the `FromRequest` equivalent of `ParseWithClaims`
	* Added new interface type `Extractor`, which is used for extracting JWT strings from http requests.  Used with `ParseFromRequest` and `ParseFromRequestWithClaims`.
	* Added several new, more specific, validation errors to error type bitmask
	* Moved examples from README to executable example files
	* Signing method registry is now thread safe
	* Added new property to `ValidationError`, which contains the raw error returned by calls made by parse/verify (such as those returned by keyfunc or json parser)

```
(Note the changes pertaining specifically to the function ParseFromRequest.)

```
	* `ParseFromRequest` has been moved to `request` subpackage and usage has changed
	* Dramatically improved the functionality and flexibility of `ParseFromRequest`, which is now in the `request` subpackage
	* Added new interface type `Extractor`, which is used for extracting JWT strings from http requests.  Used with `ParseFromRequest` and `ParseFromRequestWithClaims`.
```

So long story short, what I had to do was manually pull down the jwt-go repo specifying version v2.7.0 in order to setup uchiwa service.

**Comment by palourde** [comment]
Since we are vendoring the dependencies with Godep, adding the _-d_ flag should fix this issue: `go get -d  github.com/sensu/uchiwa`.

I'll update the documentation to reflect this change. Thank you for reporting this issue!

**Comment by mneil** [comment]
:+1: thank you both for reporting and answering this. I have been working around this for the last 30 minutes and can confirm that by just using the `-d` flag it resolves the issue.

I also believe this requires Go 1.6+ or Go 1.5 with `GO15VENDOREXPERIMENT` set to `1`. The README currently states Go 1.3 or greater required.

**Comment by umair-hussain-hs** [comment]
Glad to help!

**Comment by palourde** [comment]
@mneil We are currently using Godep and not the new vendoring feature from Golang itself so it should still work perfectly with 1.3 and greater!
